platform-sdk/swirlds-platform-core/src/main/java/com/swirlds/platform/state/address/AddressBookNetworkUtils.java

```diff
@@ -21,19 +21,42 @@
 import com.swirlds.common.system.address.Address;
 import com.swirlds.common.system.address.AddressBook;
 import com.swirlds.platform.network.Network;
+import edu.umd.cs.findbugs.annotations.NonNull;
 import java.net.InetAddress;
+import java.util.Objects;
 import org.apache.logging.log4j.LogManager;
 import org.apache.logging.log4j.Logger;
 
 /**
  * {@link AddressBook AddressBook} utility methods.
  */
-public final class AddressBookUtils {
+public final class AddressBookNetworkUtils {
 
     /** For logging info, warn, and error. */
-    private static final Logger logger = LogManager.getLogger(AddressBookUtils.class);
+    private static final Logger logger = LogManager.getLogger(AddressBookNetworkUtils.class);
 
-    private AddressBookUtils() {}
+    private AddressBookNetworkUtils() {}
+
+    /**
+     * Check if the address is local to the machine.
+     *
+     * @param address the address to check
+     * @return true if the address is local to the machine, false otherwise
+     */
+    public static boolean isLocal(@NonNull final Address address) {
+        Objects.requireNonNull(address, "The address must not be null.");
+        try {
+            return Network.isOwn(InetAddress.getByAddress(address.getAddressInternalIpv4()));
+        } catch (final Exception e) {
+            logger.error(
+                    EXCEPTION.getMarker(),
+                    "Error while checking if address {} for node {} is local",
+                    address.getAddressInternalIpv4(),
+                    address.getNodeId(),
+                    e);
+            return false;
```

Comment:
Is that really correct or should an exception be thrown?

Reply:
Initially I had both, an error in the log and an exception.   Someone made a comment asking why we need both.  I deleted the exception instead of the log message.    This method is replacing `isOwnHost` and is called where where `address.isOwnHost()` would be called and returning the same value, a boolean false.  I noticed that Exceptions seem to be dumped and not logged by executors.

Now throwing an IllegalStateException